src/components/Flipper/Flipper.tsx

```diff
@@ -261,8 +261,43 @@ function Flipper() {
             return false;
         }
         if (filter?.minVolume !== undefined) {
-            return filter?.minVolume >= 0 ? filter?.minVolume <= flipAuction.volume : Math.abs(filter?.minVolume) >= flipAuction.volume;
+            if (filter?.minVolume >= 0 && filter?.minVolume > flipAuction.volume) {
+                return false;
+            } else if (filter?.minVolume < 0 && Math.abs(filter?.minVolume) < flipAuction.volume) {
+                return false;
+            }
+        }
+
+        if (filter?.restrictions !== undefined) {
+
+            let isValid = true;
+
+            filter.restrictions.forEach(restriction => {
+
+                if (restriction.type === "blacklist" && restriction.item && restriction.item.tag === flipAuction.item.tag) {
+                    isValid = false;
```

Comment:
you should break out of the loop on the first match.
Or even better just make it a normal for loop and return false (more efficient)